Code/Editor/EditorFramework/CodeGen/CppProject.cpp

```diff
@@ -78,11 +237,161 @@ ezString ezCppProject::GetSolutionPath(const ezCppSettings& cfg)
   sSolutionFile = GetBuildDir(cfg);
-  sSolutionFile.AppendPath(cfg.m_sPluginName);
-  sSolutionFile.Append(".sln");
+
+#if EZ_ENABLED(EZ_PLATFORM_WINDOWS)
+  const ezCppProject* preferences = ezPreferences::QueryPreferences<ezCppProject>();
+  if (preferences->m_CompilerPreferences.m_eCompiler == ezCompiler::Vs2022)
+  {
+    sSolutionFile.AppendPath(cfg.m_sPluginName);
+    sSolutionFile.Append(".sln");
+    return sSolutionFile;
+  }
+#endif
+
+  sSolutionFile.AppendPath("build.ninja");
   return sSolutionFile;
 }
 
+ezStatus ezCppProject::OpenSolution(const ezCppSettings& cfg)
+{
+  const ezCppProject* preferences = ezPreferences::QueryPreferences<ezCppProject>();
+
+  switch (preferences->m_Ide.GetValue())
+  {
+#if EZ_ENABLED(EZ_PLATFORM_WINDOWS)
+    case ezIDE::VisualStudio:
+      if (!ezQtUiServices::OpenFileInDefaultProgram(ezCppProject::GetSolutionPath(cfg)))
+      {
+        return ezStatus("Opening the solution in Visual Studio failed.");
+      }
+      break;
+#endif
+    case ezIDE::VisualStudioCode:
+    {
+      auto solutionPath = ezCppProject::GetTargetSourceDir();
+      QStringList args;
+      args.push_back(QString::fromUtf8(solutionPath.GetData(), solutionPath.GetElementCount()));
+      if (ezStatus status = ezQtUiServices::OpenInVsCode(args); status.Failed())
+      {
+        return ezStatus(ezFmt("Opening Visual Studio Code failed: {}", status.m_sMessage));
+      }
+    }
+    break;
+  }
+
+  return ezStatus(EZ_SUCCESS);
+}
+
+ezStringView ezCppProject::CompilerToString(ezCompiler::Enum compiler)
+{
+  switch (compiler)
+  {
+#if EZ_ENABLED(EZ_PLATFORM_WINDOWS)
+    case ezCompiler::Vs2022:
+      return "Vs2022";
+#endif
+    case ezCompiler::Clang:
+      return "Clang";
+#if EZ_ENABLED(EZ_PLATFORM_LINUX)
+    case ezCompiler::Gcc:
+      return "Gcc";
+#endif
+    default:
+      break;
+  }
+  EZ_ASSERT_NOT_IMPLEMENTED;
+  return "<not implemented>";
+}
+
+ezCompiler::Enum ezCppProject::GetSdkCompiler()
+{
+#if EZ_ENABLED(EZ_COMPILER_CLANG)
+  return ezCompiler::Clang;
+#elif EZ_ENABLED(EZ_COMPILER_GCC)
+  return ezCompiler::Gcc;
+#elif EZ_ENABLED(EZ_COMPILER_MSVC)
+  return ezCompiler::Vs2022;
+#else
+#  error Unknown compiler
+#endif
+}
+
+ezString ezCppProject::GetSdkCompilerMajorVersion()
+{
+#if EZ_ENABLED(EZ_COMPILER_MSVC)
+  ezStringBuilder fmt;
+  fmt.Format("{}.{}", _MSC_VER / 100, _MSC_VER % 100);
+  return fmt;
+#elif EZ_ENABLED(EZ_COMPILER_CLANG)
+  return EZ_PP_STRINGIFY(__clang_major__);
+#elif EZ_ENABLED(EZ_COMPILER_GCC)
+  return EZ_PP_STRINGIFY(__GNUC__);
+#else
+#  error Unsupported compiler
+#endif
+}
+
+ezStatus ezCppProject::TestCompiler()
+{
+  const ezCppProject* preferences = ezPreferences::QueryPreferences<ezCppProject>();
+  if (preferences->m_CompilerPreferences.m_eCompiler != GetSdkCompiler())
+  {
+    return ezStatus(ezFmt("The currently configured compiler is incompatible with this SDK. The SDK was build with {} but the currently configured compiler is a {}.",
```

Comment:
Typo: "was built" instead of "was build" (also a couple more times below)
Also I think I would remove the last "a" before the brackets, and you could also put the brackets in single quotes '{}', if you want the compiler name to be more prominent.